### ISSUE TITLE
Wiki Style Links

### ISSUE THREAD

**Issue by Shackelford-Arden** [BookStackApp/BookStack#441]
So, I use SharePoint a lot at work and am very used to using the ability to easily link to other pages by simply using double open & close square brackets.

[[Page Name Here|Display Text Here]]

One of the huge benefits to me of using a method like this is that it also attempts to auto-complete you while you type based on existing pages. If a page does not exist, it adds a stub link which lets you create the page. 

While I can do this using the WYSIWIG editor, being able to do this without having to use the link button would be awesome.

**Comment by ssddanbrown** [comment]
It doesn't solve the issue but, since you've mentioned the markdown editor in another issue, That's recently received the shortcut `Ctrl+Shift+K` to show the link selector. Unfortunately it does not auto-focus on the input which would help the keyboard flow but it might still help speed things up.

**Comment by Shackelford-Arden** [comment]
Hot dang. That is indeed helpful!

I am curious about something related to this. From what I can see, these links appear to be direct links. I'm guessing that if I were to change the name of a linked page, it will likely not update the links that point to that page, correct?

Just curious from a maintenance stand point.

**Comment by Shackelford-Arden** [comment]
Also, I'll spend some time with the documentation you have... Not gonna lie, I managed to miss seeing the documentation for keyboard shortcuts.

**Comment by ssddanbrown** [comment]
@Shackelford-Arden Yeah, That's correct in regards to links being direct.

If someone changes the name of a page, Thus breaking the link, BookStack will search through revisions to match the link against past pages then take you to the new page location if found. So there is some level of link resolving. 

Will have to have a think about dynamic links and the best way to implement.

**Comment by fivestones** [comment]
I would also love the ability to both link to pages and create pages that don't yet exist by just typing [[new page|link text]] while editing, without any need to click the link button, choose the page, etc. When I want to link to something that doesn't exist yet, I would love to be able to do it, instead of having to first go make that new page, and then remember to go back and add a link to it in the page I was working on originally.

**Comment by Wachert** [comment]
would be a nice feature

**Comment by fivestones** [comment]
I'd love this feature too!

**Comment by bridgeyuwa** [comment]
Still waiting on dynamic linking

**Comment by fivestones** [comment]
Any movement on this? I’m confidant that for lots of people this could be the deciding factor on whether they go with bookstack or something else, since bookstack already has a ton of other great features.

**Comment by ssddanbrown** [comment]
@fivestones No movement on this. There'd be quite a lot of moving parts to implement and maintain for the fully functionality wished for, while introducing non-standard syntax which I generally try to steer away from.

> I’m confidant that for lots of people this could be the deciding factor on whether they go with bookstack or something else

I'm not particular concerned/focused on attracting new users since I generally prioritize existing users & uses.

**Comment by fivestones** [comment]
Ok, fair enough. But if it makes any difference, I’m one existing user whose experience would be very much improved by this ability.

**Comment by ssddanbrown** [comment]
BookStack is now [managed on Codeberg](https://codeberg.org/bookstack/bookstack).

As part of this change, we're closing all existing GitHub issues and pull requests.
You'll instead find this issue on Codeberg here: 
https://codeberg.org/bookstack/bookstack/issues/441
  
Any further discussion for this will now be done on Codeberg.
Therefore this GitHub thread will be locked.
